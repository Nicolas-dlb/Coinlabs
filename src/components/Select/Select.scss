.crypto,
.crypto2,
.currency {
  display: none;
}

#crypto-items li,
#currency-items li {
  list-style: none;
  padding-top: 5px;
  padding-bottom: 5px;
}

#crypto-items li:hover,
#currency-items li:hover {
  background: #312d49;
}

#crypto-items > li > span,
.btn-crypto > li > span,
#currency-items > li > span,
.btn-currency > li > span {
  margin-left: 30px;
  cursor: pointer;
}

/* item list */

.crypto-list {
  background: #1d1a3f;
  height: 0;
  width: 103px;
  background: #2b263f;
  transition: all 0.35s ease-in;
  border-bottom-left-radius: 15px;
  border-bottom-right-radius: 15px;
  overflow: hidden;
  cursor: pointer;
  position: absolute;
  p {
    padding-left: 11px;
  }
  transform: translateY(-25px);
  z-index: 1000;
}
.currency-list {
  @extend .crypto-list;
  transition: all 0.3s ease-in;
}

.crypto-list-active {
  height: 168px;
}
.currency-list-active {
  height: 111px !important;
  padding-top: 2px;
}

.open {
  display: flex !important;
}

.select-crypto-arrow,
.select-currency-arrow {
  width: 20px;
  position: relative;
  left: 37px;
  top: 43px;
  z-index: 101;
  cursor: pointer;
  pointer-events: none;
}

.btn-crypto {
  outline: none;
  position: relative;
  bottom: 25px;
  width: 103px;
  padding-left: 15px;
  color: var(--primary-font-color);
  font-family: Karla;
  text-align: start;
  height: 100%;
  border-radius: 15px;
  background: #2b263f;
  border: none;
  cursor: pointer;

  z-index: 100;
}

.btn-currency {
  @extend .btn-crypto;
}

.btn-crypto > li,
.btn-currency > li {
  list-style: none;
  float: center;
  position: relative;
  left: -3px;
  padding-bottom: 0px;
  cursor: pointer;
  z-index: 200;
}

@media only screen and (max-width: 992px) {
  .btn-currency {
    background: none;
  }

  .currency-list {
    background: var(--third-bg-color);
  }
}
