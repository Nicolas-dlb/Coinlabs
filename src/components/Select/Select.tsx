/* eslint-disable no-unused-vars */
/* eslint-disable no-nested-ternary */
import React, { useEffect } from "react";
import "./Select.scss";
import $ from "jquery";

type SelectProps = {
  items: string;
  setCryptoSelected?: (name: string) => void;
  setCurrencySelected?: (name: string) => void;
};

function Select({
  items,
  setCryptoSelected = undefined,
  setCurrencySelected = undefined,
}: SelectProps) {
  const langArray: any = [];
  let array = langArray;
  let type: string = "crypto";
  let secondType: string = "currency";
  let arrow: string = "select-crypto-arrow";
  let firstValue: string = "All";
  if (items === "currency") {
    const langArray2: any = [];
    array = langArray2;
    type = "currency";
    secondType = "crypto";
    arrow = "select-currency-arrow";
    firstValue = "Balance";
  }
  if (items === "crypto2") {
    const langArray3: any = [];
    array = langArray3;
    type = "crypto2";
    secondType = "crypto";
    arrow = "select-crypto2-arrow";
    firstValue = "Bitcoin";
  }

  useEffect(() => {
    // load array with items
    $(`.${type} option`).each(function loadItems() {
      const value = $(this).text();
      const item = `<li><p>${value}</p></li>`;
      if (array.length < 6) {
        array.push(item);
      }
    });
    // set options with this array

    $(`#${type}-items`).html(array);

    // Set the button value to the first element of the array
    $(`.btn-${type}`).html(array[0]);
    $(`.btn-${type}`).attr("value", `${firstValue}`);

    // change button stuff on click
    $(`#${type}-items li`).click(function selectItem(e) {
      const value: any = $(this).find("p").text();
      const item = `<li><p>${value}</p></li>`;
      $(`.btn-${type}`).html(item);
      $(`.btn-${type}`).attr("value", value);
      if (setCryptoSelected) {
        setCryptoSelected(value);
      } else if (setCurrencySelected) {
        setCurrencySelected(value);
      }

      $(`.${type}-list`).removeClass(`${type}-list-active`);
      $("#select-c").removeClass("toggle-select");
      e.stopPropagation();
    });
    // open options panel on click
    $(`.btn-${type}, .${arrow}`).click((e) => {
      $(`.${secondType}-list`).removeClass(`${secondType}-list-active`);
      $(`.${type}-list`).toggleClass(`${type}-list-active`);

      e.stopPropagation();
    });
    // close options panel when click outside
    $(document).click(() => {
      $(`.${type}-list`).removeClass(`${type}-list-active`);
      $("#select-c").removeClass("toggle-select");
    });
  }, []);

  $(`.btn-${type}, .${arrow}`).click((e) => {
    $(`.${secondType}-list`).removeClass(`${secondType}-list-active`);
    $(`.${type}-list`).toggleClass(`${type}-list-active`);

    e.stopPropagation();
  });

  return (
    <>
      <div>
        <select className={`${type} exchange_element`}>
          {items === "crypto" || items === "crypto2" ? (
            <>
              <option
                className="option exchange_element"
                label="All"
                value="All"
              >
                All
              </option>
              <option
                className="option exchange_element"
                label="Bitcoin"
                value="Bitcoin"
              >
                Bitcoin
              </option>
              <option
                className="option exchange_element"
                label="Ethereum"
                value="Ethereum"
              >
                Ethereum
              </option>

              <option
                className="option exchange_element"
                label="Ripple"
                value="Ripple"
              >
                Ripple
              </option>
              <option
                className="option exchange_element"
                label="Litecoin"
                value="Litecoin"
              >
                Litecoin
              </option>
              <option
                className="option exchange_element"
                label="Neo"
                value="Neo"
              >
                Neo
              </option>
            </>
          ) : (
            <>
              <option
                className="option exchange_element"
                label="Balance"
                value="Balance"
                id="growth_balance"
              >
                Balance
              </option>
              <option
                className="option exchange_element"
                label="Income"
                value="Income"
                id="growth_income"
              >
                Income
              </option>
              <option
                className="option exchange_element"
                label="Expenses"
                value="Expenses"
                id="growth_Expenses"
              >
                Expenses
              </option>
              <option
                className="option exchange_element"
                label="Crypto"
                value="Crypto"
                id="growth_Crypto"
              >
                Crypto
              </option>
            </>
          )}
        </select>
        <div className={`${type}-select exchange_element`}>
          <svg
            className={`select-${type}-arrow exchange_element`}
            xmlns="http://www.w3.org/2000/svg"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            fill="#605b95"
          >
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z" />
          </svg>
          <button
            id="btn-select"
            aria-label="btn-select"
            type="button"
            className={`btn-${type} exchange_element`}
            value="All"
          />
          <div className={`${type}-list exchange_element`}>
            <ul id={`${type}-items`} />
          </div>
        </div>
      </div>
    </>
  );
}

export default Select;
