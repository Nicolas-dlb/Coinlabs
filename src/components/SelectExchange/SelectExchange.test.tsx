/* eslint-disable no-unused-vars */
import { mount } from "enzyme";
import React from "react";
import Select from "./SelectExchange";

describe("Select", () => {
  let SelectComponent: any;

  beforeEach(() => {
    SelectComponent = mount(<Select items="sell" />);
  });
  afterEach(() => {
    SelectComponent = null;
  });

  it("should render correctly", () => {
    expect(SelectComponent).toMatchSnapshot();
  });

  it("should render items selected", () => {
    expect(SelectComponent.find(".sell").first().exists()).toBeTruthy();
  });

  it("should render correct options", () => {
    expect(SelectComponent.find("option").first().props().value).toBe("BTC");
  });

  // it("should show list on click", () => {
  //   SelectComponent.find(".btn-currency").simulate("click");
  //   console.log(SelectComponent.find(".currency-list").debug());
  //   expect(
  //     SelectComponent.find(".currency-list").hasClass("currency-list-active")
  //   ).toBe(true);
  // });
});
