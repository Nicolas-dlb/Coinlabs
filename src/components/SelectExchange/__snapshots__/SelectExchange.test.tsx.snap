// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `
<Select
  items="sell"
>
  <div>
    <select
      className="sell exchange_element"
    >
      <option
        className="option exchange_element"
        label="BTC"
        value="BTC"
      >
        BTC
      </option>
      <option
        className="option exchange_element"
        label="ETH"
        value="ETH"
      >
        ETH
      </option>
      <option
        className="option exchange_element"
        label="XRP"
        value="XRP"
      >
        XRP
      </option>
      <option
        className="option exchange_element"
        label="LTC"
        value="LTC"
      >
        LTC
      </option>
      <option
        className="option exchange_element"
        label="NEO"
        value="NEO"
      >
        NEO
      </option>
    </select>
    <div
      className="sell-select exchange_element"
    >
      <svg
        className="select-sell-arrow exchange_element"
        fill="#605b95"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
      <button
        aria-label="select-sell"
        className="btn-sell exchange_element"
        id="select-sell"
        type="button"
        value=""
      />
      <div
        className="sell-list exchange_element"
      >
        <ul
          id="sell-items"
        />
      </div>
    </div>
  </div>
</Select>
`;
